src/il_representations/scripts/il_train.py

```diff
@@ -190,15 +190,22 @@ def make_policy(observation_space,
             encoder = encoder_or_path
         assert isinstance(encoder, nn.Module)
 
-        # Sometimes we are not using FC layers in encoders, e.g., Jigsaw.
-        # Hence in il training we need to recover this part of the encoder.
         encoder_net = encoder.network if 'network' in dir(encoder) else \
             encoder.query_encoder.network
+
+        # Normally the last layer of an encoder is a linear layer, but in
+        # some special cases like Jigsaw, we only train the convolution
```

Comment:
Thanks for adding this explanation! Why does Jigsaw need to push the linear layer into the decoder, instead of leaving it in the encoder?

Reply:
I explained these all together in the comment above :).